optimum/intel/openvino/configuration.py

```diff
@@ -386,6 +390,13 @@ def post_init(self):
         if self.tokenizer is not None and not isinstance(self.tokenizer, str):
             raise ValueError(f"Tokenizer is expected to be a string, but found {self.tokenizer}")
 
+        if self.weight_format is None:
+            self.weight_format = "int4" if self.bits == 4 else "int8"
+        if self.weight_format not in ["int4", "int8", "mxfp4_e2m1"]:
+            raise ValueError(
+                f"Weight format must be one of the following: ['int4', 'int8', 'mxfp4_e2m1'], but found: {self.weight_format}."
+            )
```

Comment:
We already have part of this information with the bit argument so this might be a bit redundant, why not having an argument to make this distinction instead (int4 vs fp4) of introducing `weight_format` ?

Reply:
also what can/cannot be applied with mxfp4, is there any incompatibility with other arguments that we should look for here ?

Reply:
> We already have part of this information with the bit argument so this might be a bit redundant

I agree about redundancy, we had a discussion about this with @IlyasMoutawwakil above https://github.com/huggingface/optimum-intel/pull/882#discussion_r1735912690 . Is it critical in your opinion? Another option is to have `weight_format` store just `int` instead of `int4` or `int8`.

> why not having an argument to make this distinction instead (int4 vs fp4) of introducing `weight_format`?

We have other potential non-int data types for compression, for example nf4. A simple flag will do for now, but this approach will not be extendable in the future.

Reply:
> Nevertheless, I've now added corresponding checks to config post_init method. Also added similar checks for INT8 compression.

Thanks a lot for adding !



> I agree about redundancy, we had a discussion about this with @IlyasMoutawwakil above https://github.com/huggingface/optimum-intel/pull/882#discussion_r1735912690 . Is it critical in your opinion? Another option is to have weight_format store just int instead of int4 or int8.

Works for me!